### PR TITLE
Added filesystem destination docs

### PR DESCRIPTION
**Pull request by dat-a-man**

### CODE REVIEW

**Comment by dat-a-man** [comment]
@rudolfix , please verify if the permissions to read/write to s3 are correctly mentioned in the docs. I referred to this documentation: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_s3_rw-bucket.html 
But seems not to work till now. Yesterday also it took a few hours then started loading to bucket using s3fullaccess. This time I am trying a custom aws IAM policy

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# AWS S3 filesystem
+
+## Initialize the verified source
+
+1. To load the data to the AWS S3 file system, you must first initialise the project as follows:
+```
+dlt init chess filesystem
+```
+
+This command will initialise your pipeline with chess as the source and the AWS S3 filesystem as the destination.
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/destinations/filesystem.md:10]: add tabulation, 
```
1. To load the data to the AWS S3 file system, you must first initialise the project as follows:

    dlt init chess filesystem


    This command will initialise your pipeline with chess as the source and the AWS S3 filesystem as the destination.
```

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+2. Install the necessary dependencies for the S3 filesystem by running:
+
+```
+pip install -r requirements.txt
+```
+
+### Grab `credentials`
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/destinations/filesystem.md:18]: no need to highlight 'credentials' word

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+1. You can create the S3 bucket in AWS console by clicking on "Create Bucket" in S3 and assigning appropriate name and permissions to the bucket.
+2. Once the bucket is created you'll have the bucket URL. For example, If the bucket name is "mybucket_filesystem", then the bucket URL will be:
+
+```
+s3://mybucket_filesystem
+```
+
+3. To grant permissions to the user being used to access the S3 bucket, go to the IAM > Users, and click on “Add Permissions”.
+4. Add the following roles to the policy, or you can also generate your policy using [AWS policy generator](https://awspolicygen.s3.amazonaws.com/policygen.html)
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/destinations/filesystem.md:30]: dot at the end

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+2. Install the necessary dependencies for the S3 filesystem by running:
+
+```
+pip install -r requirements.txt
+```
+
+### Grab `credentials`
+
+To grab the credentials required you need to create a S3 bucket and a user who can access that bucket using certain permissions/policies.
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/destinations/filesystem.md:20]: To grab the required credentials

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md (hunk 1 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/filesystem.md:10]: done

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md (hunk 2 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/filesystem.md:18]: done

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md (hunk 3 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/filesystem.md:30]: done

File: docs/website/docs/dlt-ecosystem/destinations/filesystem.md (hunk 4 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/filesystem.md:20]: done

**Review by rudolfix — CHANGES_REQUESTED** [review]
this is missing a explanation of the file layout in the bucket and how write disposition works. this is not issue with @dat-a-man contribution. I think we need to add some depth to all our destinations on top of a walkthrough

I'll add some details here

**Comment by dat-a-man** [comment]
> this is missing a explanation of the file layout in the bucket and how write disposition works. this is not issue with @dat-a-man contribution. I think we need to add some depth to all our destinations on top of a walkthrough
> 
> I'll add some details here

Thanks @rudolfix and if you could please add some details here, I'll try updating all the other destinations with similar details 👍

**Comment by rudolfix** [comment]
@dat-a-man thanks! I rewrote a large chunk of the doc. why did you need ACL permissions to load data? only basic permissions are needed

**Comment by dat-a-man** [comment]
Thankyou @rudolfix and I'll fix that , I totally referred to the aws docs for s3 read/write policy mentioned in docs, will try again without acl permissions and update the docs :))

**Pull request merged**